plugins/modules/proxmox_backup.py

```diff
@@ -291,57 +294,28 @@ def check_storage_permissions(
             performance,
             retention):
         # Check for Datastore.AllocateSpace in the permission tree
-        if "/" in permissions.keys() and permissions["/"].get(
-                "Datastore.AllocateSpace", 0) == 1:
-            pass
-        elif "/storage" in permissions.keys() and permissions["/storage"].get("Datastore.AllocateSpace", 0) == 1:
-            pass
-        elif "/storage/" + storage in permissions.keys() and permissions["/storage/" + storage].get("Datastore.AllocateSpace", 0) == 1:
-            pass
-        else:
-            self.module.fail_json(
-                changed=False,
-                msg="Insufficient permission: Datastore.AllocateSpace is missing")
-        if (bandwidth or performance) and permissions["/"].get(
-                "Sys.Modify", 0) == 0:
-            self.module.fail_json(
-                changed=False,
-                msg="Insufficient permission: Performance_tweaks and bandwidth require 'Sys.Modify' permission for '/'")
+        if not has_permission(permissions, "Datastore.AllocateSpace", search_scopes=["/", "/storage", "/storage" + storage]):
```

Comment:
`"/storage" + storage` is missing a slash as a separator

Reply:
Like this?
```suggestion
        if not has_permission(permissions, "Datastore.AllocateSpace", search_scopes=["/", "/storage/", "/storage" + storage]):
```